### PR TITLE
feat(openapi): adapt graviti OpenAPI about branch

### PR DESCRIPTION
**Pull request by willandfree**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2j40yd7 OpenAPIv2 对接 —— Branch](https://app.clickup.com/t/2j40yd7)

File: graviti/openapi/branch.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python3
```
Comment (graczhual) [inline graviti/openapi/branch.py:1]: modify __init__, add this functions

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/branch.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                   "description": "desc",
+                   "committer": "graviti",
+                   "committed_at": "2021-03-03T18:58:10Z"
+               }
+           ],
+           "offset": 0,
+           "record_size": 1,
+           "total_count": 1
+        }
+
```
Comment (graczhual) [inline graviti/openapi/branch.py:88]: blank line

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/branch.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    """
+    url = urljoin(url, f"{URL_PATH_PREFIX}/datasets/{dataset_name}/branches")
+    params: Dict[str, Any] = {"offset": offset, "limit": limit}
```
Comment (graczhual) [inline graviti/openapi/branch.py:91]: Dict[str, int]

File: graviti/openapi/branch.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+
+def get_branch(
+    url: str,
+    access_key: str,
+    dataset_name: str,
+    *,
+    branch_name: str,
+) -> Dict[str, Any]:
+    """Execute the OpenAPI `GET /v1/datasets/{dataset_name}/branches`.
```
Comment (graczhual) [inline graviti/openapi/branch.py:102]: `GET /v1/datasets/{dataset_name}/branches/{branch_name}`

File: graviti/openapi/branch.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+    url = urljoin(url, f"{URL_PATH_PREFIX}/datasets/{dataset_name}/branches")
+    params: Dict[str, Any] = {"offset": offset, "limit": limit}
+    return open_api_do("GET", access_key, url, params=params).json()  # type: ignore[no-any-return]
+
+
+def get_branch(
+    url: str,
+    access_key: str,
+    dataset_name: str,
+    *,
```
Comment (graczhual) [inline graviti/openapi/branch.py:99]: do not need * here

**Review by graczhual — CHANGES_REQUESTED** [review]

File: graviti/openapi/branch.py (hunk 3 above)
Comment (graczhual) [reply graviti/openapi/branch.py:91]: do not need to add typing here

**Review by graczhual — APPROVED** [review]

**Review by graczhual — APPROVED** [review]

**Pull request merged**